tower-ready-cache/src/lib.rs

```diff
@@ -224,24 +224,21 @@ where
     /// Swap-removes the indexed service from the ready set.
     ///
     /// Panics if the provided `index` is out of range.
-    pub fn process_ready_index<T, P>(&mut self, index: usize, process: P) -> T
-    where
-        P: FnOnce(&mut S) -> T,
-    {
+    pub fn call_ready_index(&mut self, index: usize, req: Req) -> S::Future {
```

Comment:
I almost wonder whether the `next_ready_index` handling should also be folded into this.

Reply:
Interesting idea. We could make `poll_ready_index` part of the required API so that it is responsible for stashing the next ready index if it was ready... but I think we should _not_ make the index selection part of this module

Reply:
Sorry, yes, that's what I meant -- the index stashing, not the index _selection_.